src/deluge/model/clip/audio_clip.cpp

```diff
@@ -1122,6 +1122,29 @@ void AudioClip::setPos(ModelStackWithTimelineCounter* modelStack, int32_t newPos
 	setPosForParamManagers(modelStack, useActualPosForParamManagers);
 }
 
+bool AudioClip::shiftHorizontally(ModelStackWithTimelineCounter* modelStack, int amount) {
+	if (paramManager.containsAnyParamCollectionsIncludingExpression()) {
+		paramManager.shiftHorizontally(
+		    modelStack->addOtherTwoThingsButNoNoteRow(output->toModControllable(), &paramManager), amount, loopLength);
+	}
+	
+  int64_t newStartPos = int64_t(sampleHolder.startPos) + getSamplesFromTicks(amount);
+  if (newStartPos < 0) {
+    return false;
+  }
+  
+  uint64_t length = sampleHolder.endPos - sampleHolder.startPos;
+  
+  sampleHolder.startPos = newStartPos;
+  sampleHolder.endPos = newStartPos + length;
```

Comment:
I'm not 100% sure, but I think we need to also update `sampleholder.clustersForStart` or at least run `SampleHolder::claimClusterReasons` -- otherwise we could have an underrun when RAM is scarce.

Reply:
Ill call claimClusterReasons - i also cannot figure out exactly whether or not this is necessary, 
The sample editor calls `claimClusterReasons(reversed, CLUSTER_LOAD_IMMEDIATELY_OR_ENQUEUE)` after changing start/end points, I assume this should do the same?

Reply:
That sounds reasonable to me -- I haven't read the cluster management code in enough detail to be 100% confident but it's definitely the right shape.